app/assets/javascripts/map/components/registration-form.js

```diff
@@ -52,7 +57,11 @@ function RegistrationForm() {
           type: 'text',
           name: 'email',
           value: data.email,
-          onchange: event => { data.email = event.currentTarget.value },
+          className: alert?.message === Util.translate('registration.form.email_invalid') && 'error_input',
```

Comment:
This is some cool code, but could you use a form which is simpler for a newcomer to understand
```javascript
className: alert.status == 'error' ? 'error_input' : '',
```

Reply:
we are using alert for showing message on api failure also, wouldn't it clash with that ? like api gave error and we are setting alert.status = (some string maybe error) in that case also our field will apply class and email box will be orange.

Reply:
Good point. In another comment I suggested using `status: 'invalid'` and then you could also add another field to the alert `field: 'email'` and base it off of that. It would be overall more explicit.

Reply:
Cool, I'll add that